tests/test_debug.c

```diff
@@ -0,0 +1,74 @@
+
+#define ROM_FILE "rom:/random.dat"
+#define SD_FILE  "sd:/random.dat"
+
+#include <sys/stat.h>
+#include <unistd.h>
+
+void test_debug_sdfs(TestContext *ctx) {
+
+	if (!debug_init_sdfs("sd:/", -1)) {
+		SKIP("no SD support");
+		return;
+	}
+	DEFER(debug_close_sdfs());
+
+	struct stat info; int ret;
+	uint8_t random[8*1024];
+	FILE *randf = NULL;
+	DEFER(if (randf) fclose(randf));
+
+	// Read file from ROM
+	randf = fopen(ROM_FILE, "rb");
+	ASSERT(randf, "cannot open file: rom:/%s", ROM_FILE);
+	int sz = fread(random, 1, sizeof(random), randf);
+	ASSERT_EQUAL_UNSIGNED(sz, sizeof(random), "cannot read enough data");
+	fclose(randf); randf = NULL;
+
+	// Write file to SD.
+	randf = fopen(SD_FILE, "wb");
+	ASSERT(randf, "cannot create file: sd:/%s", SD_FILE);
+
+	// Write random-sized chunks of 1-25 bytes to stress the underlying
```

Comment:
I'm not sure what this exactly testing, Are we trying to make sure fatfs is returning the correct value? Or maybe we are trying to make sure we have correctly registered it with newlib? This might not be that valuable of a test but, will not hurt.

Reply:
The code is trying to stress `fat_disk_read_64drive` / `fat_disk_write_64drive` which have complicated logic between aligned and non-aligned reads/writes. So what I'm trying to test is the glue between FATFS and newlib. I can't be sure to cover everything because obviously it depends on fatfs internals and how aggressively it internally caches reads/writes, but I figure it out this is better than doing nothing.